### PR TITLE
Remove `_id` suffix from `BaseStorage.create_new_{study,trial}_id` methods.

### PR DESCRIPTION
**Pull request by sile**
Because these methods create not only the identifier of a new study/trial but also its instance, this PR modifies the method names to match the actual functionalities.

### CODE REVIEW

File: tests/storages_tests/rdb_tests/test_storage.py [hunk 1]
```diff
@@ -93,8 +93,8 @@ def test_create_new_study_id_multiple_studies():
     storage = create_test_storage()
     session = storage.scoped_session()
 
-    study_id_1 = storage.create_new_study_id()
-    study_id_2 = storage.create_new_study_id()
+    study_id_1 = storage.create_new_study()
```
Comment (g-votte) [inline tests/storages_tests/rdb_tests/test_storage.py]: `id` suffix still remains in the method name.

File: tests/storages_tests/rdb_tests/test_storage.py [hunk 2]
```diff
@@ -108,9 +108,9 @@ def test_create_new_study_id_duplicated_name():
 
     storage = create_test_storage()
     study_name = 'sample_study_name'
-    storage.create_new_study_id(study_name)
+    storage.create_new_study(study_name)
```
Comment (g-votte) [inline tests/storages_tests/rdb_tests/test_storage.py]: ditto

File: tests/storages_tests/test_storages.py [hunk 3]
```diff
@@ -134,7 +134,7 @@ def test_create_new_study_id_with_name(storage_mode, cache_mode):
         function_name = test_create_new_study_id_with_name.__name__
```
Comment (g-votte) [inline tests/storages_tests/test_storages.py]: ditto

File: tests/storages_tests/rdb_tests/test_storage.py (hunk 1 above)
Comment (sile) [reply tests/storages_tests/rdb_tests/test_storage.py]: Thank you for pointing it out.
I fixed the names: https://github.com/pfnet/optuna/pull/506/commits/a21678d945aca2914bd3de0e646a39a76375b9e0

**Pull request merged**